### DISCUSSION TITLE
Field validation (not null, not empty, ...) not working when the field is not provided

### DISCUSSION THREAD

**Discussion by Smirow** [directus/directus#12282] [Archive]
The field validation seems to be bypassed when the **field is not provided in the request**.
For instance this field validation:
```
{
	"testField": {
		"_nnull": true
	}
}
```
lets the user create an object with a nulled testField if it is not provided in the request payload.

### To Reproduce

- Fresh install of Directus
- Give custom permissions to the role `Public` to create `Directus_Users`
- In the custom permissions set the _field permissions_ to `first_name` and _field validation_ to `{
			"first_name": {
				"_nnull": true
			}
		}` and save

- Do a`POST` request on `/users` with an empty body, it should return code `204` and create a blank user (repeatable)

This can be reproduced with any collection.

**Comment by Smirow** [comment]
Using `_submitted` seems to solve the issue:
```
{
	"_and": [
		{
			"testField": {
				"_submitted": true
			}
		},
		{
			"testField": {
				"_nempty": true
			}
		},
		{
			"testField": {
				"_nnull": true
			}
		}
	]
}
```
Not sure if this was intended as the operator `_submitted` doesn't seem to be documented.

**Comment by rijkvanzanten** [comment]
This is technically working as it's supposed to. The validation engine validates that the submitted value matches the configured rules. It does not however require those fields to be present in the first place. That's what `_submitted` does, and that's also what the `required` flag adds on the field level 👍🏻 (cc @erondpowell for improving docs on this later)

**Reply by falmanna** [reply]
This should be documented. Took me so long to understand what is going on!
Thanks :)
